### PR TITLE
Refactor/move training files to own folder

### LINKED ISSUE #49: Separate scripts used to create the dictionaries from actual src & tests
`tests/udscore.py` and `src/dictionary_pickler.py` seem to be scripts used to create the dictionaries and validate their quality.

I suggest that they are moved to a separate folder `training` or whatever you want

### PR DESCRIPTION
**Pull request by juanjoDiaz**
Closes #49

### CODE REVIEW

**Comment by juanjoDiaz** [comment]
Once you merge #78 this is just a couple of commits

**Comment by adbar** [comment]
Could you please sort out the conflicts? Then I think we're all set.

**Comment by juanjoDiaz** [comment]
Conflicts are solved.
Still there is the problem with codecov

**Comment by adbar** [comment]
@juanjoDiaz the main code is fine but import locations changed so the tests need to be adapted.

There is also a minor mypy issue in the evaluation script which can be skipped if necessary.

**Comment by juanjoDiaz** [comment]
All fixed!

This is the last PR. I'm still writing docs for all the new stuff

File: training/eval/udscore.py [hunk 1]
```diff
@@ -66,6 +67,16 @@
...
     start = time.time()
+    dictionary_factory = DefaultDictionaryFactory()
+    strategies = DefaultStrategy(greedy=False)
+    lemmatizer = Lemmatizer(
+        dictionary_factory=dictionary_factory,
+        lemmatization_strategy=DefaultStrategy(greedy=False),
+    )
+    greedy_lemmatizer = Lemmatizer(
+        dictionary_factory=dictionary_factory,
+        lemmatization_strategy=DefaultStrategy(greedy=False),
```
Comment (adbar) [inline training/eval/udscore.py:78]: `greedy=True` maybe?

**Pull request merged**